src/main/java/net/minestom/server/particle/data/VibrationBlockParticleData.java

```diff
@@ -0,0 +1,18 @@
+package net.minestom.server.particle.data;
+
+import net.minestom.server.coordinate.Point;
+import net.minestom.server.network.NetworkBuffer;
+import org.jetbrains.annotations.NotNull;
+
+public record VibrationBlockParticleData(Point source, int ticks) implements ParticleData {
```

Comment:
I'm not super convinced on these being two different records, they do correspond to the same packet. For other packets with unions like this they are generally in a single record